### PR TITLE
added secret manager logic

### PR DESCRIPTION
**Pull request by vpchinta**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 1]
```diff
@@ -187,6 +193,17 @@ private Incident updateIncident(Context context, Incident incident) {
 
     // An HttpRequest instance is built through an HttpRequest builder
     private HttpRequest.Builder getBuilderWithBasicValues(String uriPath) {
+
+        JSONObject serviceNowCredentials = getSecret(secretName);
+
+        String username = new String();
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java]: Only declare the variables.  ```String userName```. In Java, there is no need to create empty String Objects.

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 2]
```diff
@@ -187,6 +193,17 @@ private Incident updateIncident(Context context, Incident incident) {
...
+        String password = new String();
+        try {
+            username = serviceNowCredentials.getString("username");
+            password = serviceNowCredentials.getString("password");
+        } catch (JSONException e) {
+            throw new RuntimeException(e);}
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java:205]: Log the exception.

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 3]
```diff
@@ -25,8 +32,7 @@ public class ServiceNowConnector {
     public static final String NEW_LINE = System.lineSeparator();
 
     // Configure your Environment variables in AWS Lambda.
-    public static String username = System.getenv("USER_NAME");
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java:28]: Are you removing the option to pass user name and password ?. Have you confirmed this with Blog writers ?

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 4]
```diff
@@ -326,4 +344,30 @@ public static JsonNode getJsonNodeFromInput(Map<String, Object> input, final Con
...
     }
-}
\ No newline at end of file
+
+    public JSONObject getSecret(String secretName) {
+
+        AWSSecretsManager client = AWSSecretsManagerClientBuilder.standard()
+                .build();
+
+        String secret = "";
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java]: Only declaration of ```String secret;``` is sufficient.

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 5]
```diff
@@ -326,4 +344,30 @@ public static JsonNode getJsonNodeFromInput(Map<String, Object> input, final Con
...
+        GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest()
+                .withSecretId(secretName);
+        GetSecretValueResult getSecretValueResult = null;
+
+        try {
+            getSecretValueResult = client.getSecretValue(getSecretValueRequest);
+        } catch (Exception e) {
+            throw e;
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java:362]: log the exception.

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java [hunk 6]
```diff
@@ -326,4 +344,30 @@ public static JsonNode getJsonNodeFromInput(Map<String, Object> input, final Con
...
+        }
+        if (getSecretValueResult.getSecretString() != null) {
+            secret = getSecretValueResult.getSecretString();
+        }
+        try {
+            JSONObject jsonObject = new JSONObject(secret);
+            return jsonObject;
+        } catch (JSONException e) {
+            throw new RuntimeException(e);
```
Comment (kiruba-sampath) [inline Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java]: Log the exception.

**Review by kiruba-sampath — COMMENTED** [review]
Please address the comments before merging.

File: Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java (hunk 3 above)
Comment (vpchinta) [reply Functions/src/main/java/aws/devopsguru/partner/servicenow/ServiceNowConnector.java:28]: Yes we need to remove secret name and password, as main purpose of using secret manager is to protect our credentials. Will discuss it with blog writers.

**Review by kiruba-sampath — APPROVED** [review]

**Review by kiruba-sampath — APPROVED** [review]

**Review by kiruba-sampath — APPROVED** [review]

**Review by kiruba-sampath — APPROVED** [review]

**Pull request merged**